platform/pulp/platform/models/repository.py

```diff
@@ -213,44 +225,20 @@ class Importer(Plugin):
 
     scratchpad = fields.GenericRelation(Scratchpad)
 
-    class Meta:
-        abstract = True
-
-
-class RepositoryImporter(Importer):
-    """
-    A content importer that is associated with a repository.
-
-    Fields:
-
-    Relations:
-
-    :cvar repository: The associated repository.
-    :type repository: models.ForeignKey
-    """
-
     repository = models.ForeignKey(
-        Repository, related_name='importers', on_delete=models.CASCADE)
-
-    class Meta:
-        unique_together = ('repository', 'name')
-
-    def natural_key(self):
-        """
-        Get the model's natural key.
-
-        :return: The model's natural key.
-        :rtype: tuple
-        """
-        return (self.repository.id, self.name)
+        Repository, blank=True, null=True, related_name='importers', on_delete=models.CASCADE)
 
 
-class Distributor(Plugin):
+class Publisher(ContentAdaptor):
     """
-    An abstract content distributor.
+    A content publisher.
 
     Fields:
 
+    :cvar auto_publish: Indicates that the adaptor may publish automatically
```

Comment:
IIRC, I chose `may` because I wanted to leave the door open for a publisher to make the final decision to actually publish.  However, each publisher with auto-publish=True `will` be requested to publish.  I can make this clearer.  Or, I also fine with changing this to `will` here.